test/LargeStorageTest.t.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+import "contracts/FlatDirectory.sol";
+
+contract FlatDirectoryTest is Test {
+    FlatDirectory fd;
+
+    function setUp() public {
+        fd = new FlatDirectory(0, 0, address(0));
+    }
+
+    function testReadWrite() public {
+        bytes memory name = hex"616263";
+        bytes memory data = hex"112233";
+        fd.write(name, data);
+
+        (bytes memory result, bool ok) = fd.read(name);
+        assertTrue(ok);
+        assertEq(result, data);
+
+        bytes memory randomData = _randBytes(40);
+        fd.write(name, randomData);
+
+        (result, ok) = fd.read(name);
+        assertTrue(ok);
+        assertEq(result, randomData);
+
+        (uint256 size, uint256 count) = fd.size(name);
+        assertEq(size, 40);
+        assertEq(count, 1);
+    }
+
+    function testReadWriteChunks() public {
+        bytes memory name = hex"616263";
+        bytes memory chunk0 = _randBytes(1024);
+        fd.write(name, chunk0);
+
+        (bytes memory result, bool ok) = fd.read(name);
+        assertTrue(ok);
+        assertEq(result, chunk0);
+
+        bytes memory chunk1 = _randBytes(512);
+        fd.writeChunk(name, 1, chunk1);
+
+        (result, ok) = fd.readChunk(name, 1);
+        assertTrue(ok);
+        assertEq(result, chunk1);
+
+        bytes memory concat = bytes.concat(chunk0, chunk1);
+        (result, ok) = fd.read(name);
+        assertTrue(ok);
+        assertEq(result, concat);
+
+        (uint256 size, uint256 count) = fd.size(name);
+        assertEq(size, 1536);
+        assertEq(count, 2);
+    }
+
+    function testWriteRemoveChunks() public {
+        bytes memory name = hex"616263";
+        assertEq(fd.countChunks(name), 0);
+
+        bytes memory chunk0 = _randBytes(10);
+        fd.write(name, chunk0);
+
+        (bytes memory result, bool ok) = fd.read(name);
+        assertTrue(ok);
+        assertEq(result, chunk0);
+
+        bytes memory chunk1 = _randBytes(20);
+        fd.writeChunk(name, 1, chunk1);
+
+        (result, ok) = fd.readChunk(name, 1);
+        assertTrue(ok);
+        assertEq(result, chunk1);
+
+        fd.removeChunk(name, 0);
```

Comment:
we may also need to check the return value

Reply:
fix